server/schedulers/balance_leader.go

```diff
@@ -71,15 +80,106 @@ func init() {
 }
 
 type balanceLeaderSchedulerConfig struct {
-	Name   string          `json:"name"`
-	Ranges []core.KeyRange `json:"ranges"`
-	Batch  int             `json:"batch"`
+	mu      sync.RWMutex
+	storage endpoint.ConfigStorage
+	Name    string          `json:"name"`
```

Comment:
how about remove Name?

Reply:
I want to abstract `BaseSchedulerConfig` like `BaseScheduler` in the future, it will use `Name` when persisting config

Reply:
Does that mean we can change the name through API?

Reply:
we can't change the name. I will remove it.